lib/enum/basic_enum.rb

```diff
@@ -71,6 +81,7 @@ def drop_until_its_hot
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_slice
   # Tip: After .each_slice chain on .to_a to turn the returned Enumerator into an array
   def groups_of(num)
+    collection.each_slice(num).map { |a| p a }.to_a
```

Comment:
You don't need the .map in there at all, and you definitely shouldn't be using p, which is a shortcut for puts/print. `collection.each_slice(num).to_a` will get you the same exact result.